crates/transpiler/src/passes/disjoint_layout.rs

```diff
@@ -0,0 +1,545 @@
+            let cmap = subgraph(
+                &coupling_map,
+                &qubits.iter().map(|x| NodeIndex::new(x.index())).collect(),
+            );
+            Ok(Some(vec![func(dag.clone(), &cmap)?]))
+        }
+        DisjointSplit::Arbitrary(components) => Some(
+            components
+                .into_iter()
+                .map(|component| {
+                    let coupling_map = build_coupling_map(target).unwrap();
+                    let cmap = subgraph(
+                        &coupling_map,
+                        &component
+                            .physical_qubits
+                            .iter()
+                            .map(|x| NodeIndex::new(x.index()))
+                            .collect(),
+                    );
+                    func(component.sub_dag, &cmap)
+                })
+                .collect::<PyResult<Vec<_>>>(),
+        )
+        .transpose(),
+    }
+}
+
+pub fn distribute_components(dag: &mut DAGCircuit, target: &Target) -> PyResult<DisjointSplit> {
+    let coupling_map: CouplingMap = match build_coupling_map(target) {
+        Some(map) => map,
+        None => return Ok(DisjointSplit::NoneNeeded),
+    };
+    let cmap_components = connected_components(&coupling_map);
+    if cmap_components.len() == 1 {
+        if dag.num_qubits() > cmap_components[0].len() {
+            return Err(TranspilerError::new_err(concat!(
+                "A connected component of the DAGCircuit is too large for any of the connected ",
+                "components in the coupling map."
+            )));
+        }
+        return Ok(DisjointSplit::NoneNeeded);
+    }
+    let dag_components = separate_dag(dag)?;
+    let mapped_components = map_components(&dag_components, &cmap_components)?;
+    let out_component_pairs: Vec<(DAGCircuit, CouplingMap)> = mapped_components
+        .into_iter()
+        .enumerate()
+        .filter(|(_, dag_indices)| !dag_indices.is_empty())
+        .map(|(cmap_index, dag_indices)| {
+            let mut out_dag = dag_components[*dag_indices.first().unwrap()].clone();
+            for dag_index in &dag_indices[1..] {
+                let dag = &dag_components[*dag_index];
+                for qubit in dag.qubits().objects() {
+                    out_dag.add_qubit_unchecked(qubit.clone())?;
+                }
+                for clbit in dag.clbits().objects() {
+                    out_dag.add_clbit_unchecked(clbit.clone())?;
+                }
+                for qreg in dag.qregs() {
+                    out_dag.add_qreg(qreg.clone())?;
+                }
+                for creg in dag.cregs() {
+                    out_dag.add_creg(creg.clone())?;
+                }
+                out_dag.compose(
+                    dag,
+                    Some(dag.qubits().objects()),
+                    Some(dag.clbits().objects()),
+                    false,
+                )?;
+            }
```

Comment:
This was pre-existing so we can leave it, but I think the call to `compose` here would cause errors if there are `Var` or `Stretch` instances in `dag`.

Reply:
Should I change `inline_captures` to `true` or do I need to do more work here? It's unlikely for this to come up in practice but it would still be good to fix this.

Reply:
I think the fix would be changing each `dag` component to be created from an initial empty-like template that sets `vars_mode="captures"`, then we'd change this call to `compose` to have `inline_captures=true`, but let's just follow up at some later point rather than trying to change it on-the-fly.